.travis.yml

```diff
@@ -23,10 +37,22 @@ before_install:
 #            - texlive-fonts-recommended
 #            - texlive-fonts-extra
 #            - dvipng
-install: python setup.py install
+
+install:
+    - ./.travis/install.sh
+    - if ! [[ $TRAVIS_OS_NAME == "linux" ]]; then source $HOME/miniconda/bin/activate; fi
+    - if ! [[ $TRAVIS_OS_NAME == "linux" ]]; then hash -r ; fi
+    - if ! [[ $TRAVIS_OS_NAME == "linux" ]]; then source activate test-environment ; fi
+    - export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib
+    - export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/lib/pkgconfig
+    - cmake --version
+    - python --version
+
 script:
-    - python tests/run_tests.py --debug --detailed-errors --verbose --process-restartworker
-    - make check-cpp
+    - python setup.py build
+    - python setup.py test
```

Comment:
Some of the tests are decorated with a `slow` tag, so that they can be ignored by default when users run tests on their local machine. We want them to be run by Travis though. Is this what the `test` target does, or do we need to add a `--all` argument or similar?

Reply:
good point, looks like doing python setup.py test just runs everything (even the slow tests). To have more control of the tests, one could use the `python setup.py nosetests` command (from what I understand `numpy.testing.run_module_suite` is just a wrapper for nosetests)